users/tests/test_api_user.py

```diff
@@ -0,0 +1,222 @@
+from django.test import Client
+from django.urls import reverse
+from rest_framework.exceptions import ErrorDetail
+from rest_framework.test import APIClient
+
+User = get_user_model()
+
+
+@pytest.fixture()
+def api_user_url(user: User) -> str:
+    return reverse("api-user", kwargs={"nickname": user.nickname})
+
+
+@pytest.fixture()
+def api_user_profile_url() -> str:
+    return reverse("api-user-profile")
+
+
+@pytest.fixture()
+def api_user_settings_url() -> str:
+    return reverse("api-user-settings")
+
+
+@pytest.mark.django_db()
+def test_api_user_valid_data_response_success(
+    authenticated_client: tuple[APIClient, str, str],
+    api_user_url: str,
+) -> None:
+    client, _, _ = authenticated_client
+    response = client.get(api_user_url)
+    assert response.status_code == 200
+    assert response.data["nickname"] == "test_user"
+    assert response.data["email"] == "test@example.com"
+
+
+@pytest.mark.django_db()
+def test_api_user_invalid_nickname_response_failed(
+    authenticated_client: tuple[APIClient, str, str],
+) -> None:
+    client, _, _ = authenticated_client
+    response = client.get(reverse("api-user", kwargs={"nickname": "test"}))
+    assert response.status_code == 404
+
+
+@pytest.mark.django_db()
+def test_api_user_unauthenticated_user_response_failed(
+    client: Client, user_credentials: tuple[APIClient, str, str], api_user_url: str
+) -> None:
+    response = client.get(api_user_url)
+    assert response.status_code == 401
+    assert isinstance(response.data["detail"], ErrorDetail)
+
+
+@pytest.mark.django_db()
+def test_api_user_profile_valid_data_response_success(
+    authenticated_client: tuple[APIClient, str, str], api_user_profile_url: str, user: User
+) -> None:
+    client, _, _ = authenticated_client
+    response = client.get(api_user_profile_url)
+    assert response.status_code == 200
+    assert response.data == {
+        "id": user.profile.id,
+        "bio": user.profile.bio,
+        "is_nsfw": user.profile.is_nsfw,
+        "is_followable": user.profile.is_followable,
+        "is_content_visible": user.profile.is_content_visible,
+        "is_communities_visible": user.profile.is_communities_visible,
+        "comment_karma": user.profile.comment_karma,
+        "post_karma": user.profile.post_karma,
+        "gold_awards": user.profile.gold_awards,
+        "gender": user.profile.gender,
+        "avatar": user.profile.avatar,
+        "banner": user.profile.banner,
+        "user": user.id,
+    }
+
+
+@pytest.mark.django_db()
+def test_patch_api_user_profile_valid_data_response_success(
+    authenticated_client: tuple[APIClient, str, str], api_user_profile_url: str, user: User
+) -> None:
+    client, _, _ = authenticated_client
+    data = {"is_followable": not user.profile.is_followable}
+    response = client.patch(api_user_profile_url, data=data)
+    assert response.status_code == 200
+    assert response.data == {
+        "id": user.profile.id,
```

Comment:
Just a remark, about testing. 
As a good practice it is better to use values than variables in assertions in testing. Of course, it varies case by case. But, for exampe adding a string inot "bio" like
```
"bio" : "this is example user bio, lorem ipsum etc"
```
shows more detail, not assumptions. You can clearly see it is a string, not null for example. 
I am not saying anything is to change here. Just saying to always consider using real values, not variables, cause it helps in profile readability. 
It may lower a bit test maintainability, though, so it's always a tradeoff.